### DISCUSSION TITLE
How do I get the auto_id when I insert a vector?

### DISCUSSION THREAD

**Discussion by Ysnower** [milvus-io/milvus#31422] [Q&A and General discussion]
How do I get the auto_id when I insert a vector?

**Comment by yhmo** [comment]
If the insert() interface succeed, it will return the auto-generated ids:
```
ret = collection.insert(data)
print("insert done", ret.primary_keys)
```

**Reply by Ysnower** [reply]
Thanks for your reply!

**Reply by JenuelDev** [reply]
@Ysnower Hi, how do auto create id? because when I am inserting an entities format like this:

```
[
 { source: "some text here", embeddings: [1,2,3,4,5] },
 ...
]
```

but its throwing an error:
<ParamError: (code=1, message=Field id don't match in entities[0])>

**Reply by yhmo** [reply]
Assume the collection has 3 fields:
```
fields=[
    FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name="embeddings", dtype = DataType.FLOAT_VECTOR, dim=5),
    FieldSchema(name="source", dtype = DataType.VARCHAR, max_length=100),
]
```
"id" is primary key, auto_id=True

Row-based insert:
`collection.insert(data = {"source": "aaa", "embeddings": [1, 2, 3, 4, 5]})   # insert one row, no need to provide "id"
 
Column-based insert:
```
data = [
  ["aaa", "bbb", "ccc"], # this is for "source", 3 rows
  [[1, 2, 3, 4, 5], [2, 3, 4, 5, 6], [3, 4, 5, 6, 7]]     # this is for "embeddings", 3 rows
]
collection.insert(data=data) # no need to provide "id"
```

Rich examples in pymilvus repo: https://github.com/milvus-io/pymilvus/tree/master/examples
